packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -1220,6 +1378,47 @@ class MemoryTimeline {
   }
 }
 
+/// Compute a simple moving average.
+class MovingAverage {
+  MovingAverage({this.averagePeriod = 20, this.magnitudeSpike = 3 });
+
+  final dataSet = Queue<int>();
+
+  // Total collected items in the X axis (time) used to compute moving average.
+  final int averagePeriod;    // Default 10-20 seconds.
+
+  final int magnitudeSpike;   // Default magnitude of comparing first to mean.
+
+  // Sum of total heap used and external heap for unitPeriod.
+  int averageSum = 0;
+
+  // Reset moving average data.
+  void clear() {
+    dataSet.clear();
+    averageSum = 0;
+  }
+
+  // Update the sum to get a new mean.
+  void addToMA(int value) {
```

Comment:
nit: rename to just `add` or `addToMovingAverage` if `add` isn't clear enough.

Reply:
Done.